### PR TITLE
ENI Support

### PR DESCRIPTION
**Pull request by sameshai**

### CODE REVIEW

File: lib/provider/vpc_data_types.go [hunk 1]
```diff
@@ -50,6 +54,20 @@ type VPC struct {
...
 
+// SecurityGroup ...
+type SecurityGroup struct {
+	ID   string `json:"id"`
+	CRN  string `json:"crn,omitempty"`
+	Href string `json:"href,omitempty"`
+}
+
+// PrimaryIP ...
+type PrimaryIP struct {
```
Comment (sameshai) [inline lib/provider/vpc_data_types.go:65]: HREF can be added

File: lib/provider/vpc_data_types.go [hunk 2]
```diff
@@ -50,6 +54,21 @@ type VPC struct {
...
+	ID      string `json:"id,omitempty"`
+	Href    string `json:"href,omitempty"`
```
Comment (sameshai) [inline lib/provider/vpc_data_types.go:67]: Lets have logical seperation in ID and address

**Review by prankulmahajan — APPROVED** [review]
lgtm

File: lib/provider/vpc_data_types.go (hunk 1 above)
Comment (arahamad) [reply lib/provider/vpc_data_types.go:78]: Can you add href for all resource struct

File: lib/provider/volume_file_access_point.go [hunk 3]
```diff
@@ -59,13 +65,57 @@ type VolumeAccessPointRequest struct {
...
+	//ResourceGroup for ENI
+	ResourceGroup *ResourceGroup `json:"resource_group,omitempty"`
+
+	//EncryptionInTransit
+	EncryptionInTransit string `json:"transit_encryption,omitempty"`
 }
 
-//VolumeAccessPointResponse used for both delete and create access point
+// VolumeAccessPointResponse used for both delete and create access point
 type VolumeAccessPointResponse struct {
```
Comment (arahamad) [inline lib/provider/volume_file_access_point.go:86]: Can you add href for all resource struct

File: lib/provider/vpc_data_types.go [hunk 4]
```diff
@@ -50,6 +54,32 @@ type VPC struct {
...
+// SecurityGroup ...
+type SecurityGroup struct {
+	ID   string `json:"id"`
+	CRN  string `json:"crn,omitempty"`
+	Href string `json:"href,omitempty"`
+}
+
+// PrimaryIPID ...
+type PrimaryIPID struct {
+	ID   string `json:"id,omitempty"`
```
Comment (arahamad) [inline lib/provider/vpc_data_types.go:66]: Please check if ID and href can be omited

**Review by arahamad — APPROVED** [review]
Please fix the review comments in next PR

**Pull request merged**